### PR TITLE
migrateImagesBetweenSwiftDC: check for empty files

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/PLATFORM-841
- avoid `HTTP 411` responses from Swift by checking if the source file is not empty
- do not retry moves of empty files
- log all operations to LogStash

@michalroszka / @wladekb / @ljagiello

### CODE REVIEW

File: maintenance/wikia/migrateImagesBetweenSwiftDC.php [hunk 1]
```diff
@@ -32,47 +32,49 @@ public function __construct() {
...
 	}
-	
+
 	/**
 	 * Create container and authenticate - for destination Ceph/Swift storage
 	 *
-	 * @return SwiftStorage storage instance
+	 * @return Wikia\SwiftStorage storage instance
 	 */
 	private function destConn() {
```
Comment (macbre) [inline maintenance/wikia/migrateImagesBetweenSwiftDC.php]: `s/$this->imageSyncQueue->city_id/$city_id/g'

File: maintenance/wikia/migrateImagesBetweenSwiftDC.php [hunk 2]
```diff
@@ -12,17 +12,17 @@
...
 	/* @private Int - number of images to sync */
 	private $mLimit;
 	/* @private String - destination DC */
 	private $mDC_dst;
 	/* @private Array - containers */
-	private $source_container;
-	private $dest_container;
-	/* @private \Queue object  */
+	private $source_container = [];
+	private $dest_container = [];
```
Comment (michalroszka) [inline maintenance/wikia/migrateImagesBetweenSwiftDC.php]: As a sidenote: `src` and `dst` prefixes work better and most engineers should be familiar with them as `tcpdump` uses them, too :-)

File: maintenance/wikia/migrateImagesBetweenSwiftDC.php [hunk 3]
```diff
@@ -201,32 +226,36 @@ private function store() {
...
+	 * @return Boolean|null return false will re-queue the item, null - skip it
+	 */
 	private function delete() {
 		$this->output( "\tDelete {$this->imageSyncQueue->dst} image\n" );
-		
+
 		/* connect to source Ceph/Swift */
 		$srcStorage = $this->srcConn();
-		
+
```
Comment (michalroszka) [inline maintenance/wikia/migrateImagesBetweenSwiftDC.php]: Oh, here we use `src` and `dst`.

File: maintenance/wikia/migrateImagesBetweenSwiftDC.php (hunk 2 above)
Comment (macbre) [inline maintenance/wikia/migrateImagesBetweenSwiftDC.php]: True, true - however, I don't want to rewrite the whole script :) These are just minor improvements of errors handling and logging.

**Comment by michalroszka** [comment]
LGTM. My comments on style and convention are suggestions for the future rather than criticisms.

File: includes/wikia/SwiftStorage.class.php [hunk 4]
```diff
@@ -104,7 +104,7 @@ private function __construct( $containerName, $pathPrefix = '', $dataCenter = nu
 	 */
 	protected function getLoggerContext() {
 		return [
-			'container'    => $this->getContainerName(),
+			'container'    => $this->getContainerName() . $this->getPathPrefix(), // eg. "poznan" + "/pl/images"
```
Comment (wladekb) [inline includes/wikia/SwiftStorage.class.php]: If it's no longer a container, maybe we should rename the key?

File: maintenance/wikia/migrateImagesBetweenSwiftDC.php [hunk 5]
```diff
@@ -111,29 +113,43 @@ public function execute() {
 				} else {
 					$res = $this->store();
 				}
-				
+
+				/**
+				 * $res == true  - we're fine
```
Comment (wladekb) [inline maintenance/wikia/migrateImagesBetweenSwiftDC.php]: === ?

File: maintenance/wikia/migrateImagesBetweenSwiftDC.php [hunk 6]
```diff
@@ -111,29 +113,43 @@ public function execute() {
...
+				 * $res == false - error, keep an item in the queue (i.e. retry in the next run)
+				 * $res == null  - error, move the item to archive (i.e. ignore the error)
+				 */
 				if ( $res === null ) {
 					$this->output( "\tFile ({$this->imageSyncQueue->dst}) doesn't exist in source DC\n" );
 					$error = 5;
```
Comment (wladekb) [inline maintenance/wikia/migrateImagesBetweenSwiftDC.php]: Would be nice to have a constant here.

File: maintenance/wikia/migrateImagesBetweenSwiftDC.php [hunk 7]
```diff
@@ -141,22 +157,22 @@ private function getRemotePath( $path ) {
...
 	/**
 	 * Store image in destination path
 	 *
 	 * @param $city_id Int - Wikia ID
 	 * @param $src String - source file
 	 * @param $dst String - destination path
-	 * 
-	 * @return Boolean
+	 *
+	 * @return Boolean|null return false will re-queue the item, null - skip it
```
Comment (wladekb) [inline maintenance/wikia/migrateImagesBetweenSwiftDC.php]: I'd rather write it from the function perspective (not how is later used) like:
Returns false for recoverable error and null for permanent error.

File: maintenance/wikia/migrateImagesBetweenSwiftDC.php (hunk 5 above)
Comment (macbre) [inline maintenance/wikia/migrateImagesBetweenSwiftDC.php]: Come on, it's a code comment ;)

Comment (wladekb) [inline maintenance/wikia/migrateImagesBetweenSwiftDC.php]: Yet, it is misleading.

**Pull request merged**